example/generators/event_gen.cpp

```diff
@@ -0,0 +1,179 @@
+/*
+ * A miniapp that demonstrates how to use event generators.
+ *
+ * The miniapp builds a simple model of a single cell, with one compartment
+ * corresponding to the soma. The soma has a single synapse, to which two
+ * event generators, one inhibitory, and one excitatory, are attached.
+ */
+
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+
+#include <json/json.hpp>
+
+#include <cell.hpp>
+#include <common_types.hpp>
+#include <event_generator.hpp>
+#include <hardware/node_info.hpp>
+#include <load_balance.hpp>
+#include <model.hpp>
+#include <recipe.hpp>
+#include <simple_sampler.hpp>
+
+using arb::cell_gid_type;
+using arb::cell_lid_type;
+using arb::cell_size_type;
+using arb::cell_member_type;
+using arb::cell_kind;
+using arb::time_type;
+using arb::cell_probe_address;
+
+// Writes voltage trace as a json file.
+void write_trace_json(const arb::trace_data<double>& trace);
+
+class generator_recipe: public arb::recipe {
+public:
+    // There is just the one cell in the model
+    cell_size_type num_cells() const override {
+        return 1;
+    }
+
+    // Create cell with just a single compartment for the soma:
+    //    soma diameter: 18.8 µm
+    //    mechanisms: pas [default params]
+    //    bulk resistivitiy: 100 Ω·cm [default]
+    //    capacitance: 0.01 F/m² [default]
+    //    synapses: 1 * expsyn
+    arb::util::unique_any get_cell_description(cell_gid_type gid) const override {
+        arb::cell c;
+
+        c.add_soma(18.8/2.0);
```

Comment:
Magic value 2.0

Reply:
The comment above states that the soma has diameter 18um. The argument is a radius.